crates/flux-tests/tests/neg/surface/maps00.rs

```diff
@@ -0,0 +1,49 @@
+#![feature(register_tool)]
+#![register_tool(flux)]
+#![feature(custom_inner_attributes)]
+#![flux::defs {
+    fn map_set(m:Map<int, int>, k: int, v: int) -> Map<int, int> { map_store(m, k, v) }
+    fn map_get(m: Map<int,int>, k:int) -> int { map_select(m, k) }
+    fn map_def(v:int) -> Map<int, int> { map_default(v) }
+}]
+
+#[flux::sig(fn (bool[true]))]
+fn assert(_b: bool) {}
+
+/// define a type indexed by a map
+#[flux::opaque]
+#[flux::refined_by(map: Map<int, int>)]
+struct MyMap {
+    inner: std::collections::HashMap<i32, i32>,
+}
+
+impl MyMap {
+    #[flux::trusted]
+    pub fn new() -> Self {
+        Self { inner: std::collections::HashMap::new() }
+    }
+
+    #[flux::trusted]
+    #[flux::sig(fn(self: &strg MyMap[@m], k: i32, v: i32) ensures self: MyMap[map_set(m, k, v)])]
+    pub fn set(&mut self, k: i32, v: i32) {
+        self.inner.insert(k, v);
+    }
+
+    #[flux::trusted]
+    #[flux::sig(fn(&MyMap[@m], k: i32) -> Option<i32[map_get(m, k)]>)]
```

Comment:
Can we do something like `fn(&MyMap[@m], k: i32{contains(m, k)}) -> i32[map_get(m, k)]`? Also, what's the semantics of `map_get` when the key is not in the map? Does it return an unspecified value?

Reply:
in the smt theory you get some arbitrary value -- the map is a "total" function from dom->rng. 

we could have a `contains` thing if we separately tracked the `Set` of keys... let me try that.

Reply:
or if we had an `option<T>` sort I suppose, where the `new` map returns a `Map_default(none)`.